exercises/chapter 3/chapter3.Rmd

```diff
@@ -0,0 +1,179 @@
+```
+
+The `print()` function in R is used to display the values of variables or expressions on the screen. Unlike directly calling the variable, using `print()` provides explicitness and allows you to control the output format when displaying values.
+
+#### Test Your Code (Exercise 1)
+
+Do not modify the cell below. Run the cell below to check if you've defined and assigned the variables correctly.
+
+```{r}
+# Load testthat package
+library(testthat)
+
+# Unittest for Exercise 1
+test_that("Exercise 1: Variables and Assignments", {
+  
+  expect_that(my_age, is_a("numeric"))
+  expect_gte(my_age, 0)
+  expect_lte(my_age, 100)
+  
+  expect_that(my_name, is_a("character"))
+  expect_gt(length(my_name), 0)
+  expect_lte(length(my_name), 50)
+  
+})
+```
+
+------------------------------------------------------------------------
+
+# Exercise 2: Installing Packages
+
+**Your Task:**\
+1. Install the `xgboost` package.\
+2. Install `xml` and `mlr3` packages in one line of code by passing in two package names as a single variable.
+
+**Consider:** You might want to run `?install.packages()` to see what data type it allows for the package names.
+
+```{r}
+# Complete the code below and run the cell.
+
+# Install xgboost -- your code here
+# Install xml and mlr3 -- your code here
+```
+
+**What are these packages?**\
+`xgboost` is an efficient and scalable machine learning library designed for gradient boosting. It's widely used for supervised learning tasks, especially in structured/tabular data, due to its high performance and accuracy.
+
+The `xml` package in R provides tools to parse and manipulate XML (eXtensible Markup Language) files, commonly used for data exchange between different systems and applications.
+
+`mlr3` is an R package for machine learning in R that provides a unified interface and infrastructure for various machine learning tasks such as data preprocessing, model training, hyperparameter tuning, and evaluation. It's a successor to the popular mlr package and is designed to be more modular and extendable.
+
+#### Test Your Code (Exercise 2)
+
+Do not modify the cell below. Run the cell below to check if you've successfully installed the packages.
+
+```{r}
+library(testthat)
+test_that("Installing Packages", {
+  # Task 1: Check if tidyverse is installed
+  expect_true("xgboost" %in% installed.packages()[, "Package"])
+  
+  # Task 2: Check if ggplot2 and dplyr are installed
+  packages_installed <- c("xml", "mlr3")
+  expect_true(all(packages_installed %in% installed.packages()[, "Package"]))
+})
+```
+
+------------------------------------------------------------------------
+
+# Exercise 3: Basic Calculation
+
+**Your Task:**\
+1. Add 25 and 37, then divide it by 2. Assign it to a variable `A`.\
+2. Multiply `A` by 5, then subtract it by pi(π). Assign it to a variable `B`.\
+3. Raise `B` to the power of 5. Assign it to a variable "C".\
+4. Test if C is an integer. Assign the test result to a variable "C_is_integer".
```

Comment:
I can't remember if the chapter covers the term "boolean", but if it does then please use that here. And if it doesn't, say explicitly that the variable's value should be either `TRUE` or `FALSE`.